app/user.go

```diff
@@ -1,33 +1,61 @@
 package app
 
-import "github.com/ubclaunchpad/cumulus/blockchain"
+import (
+	"github.com/ubclaunchpad/cumulus/blockchain"
+
+	crand "crypto/rand"
+)
 
 // User holds basic user information.
 type User struct {
-	HotWallet
+	*blockchain.Wallet
 	BlockSize uint32
 }
 
-// HotWallet is a representation of the users wallet.
-type HotWallet struct {
-	Name string
-	blockchain.Wallet
-}
-
 // NewUser creates a new user
 func NewUser() *User {
 	return &User{
-		HotWallet: HotWallet{
-			Wallet: blockchain.NewWallet(),
-			Name:   "default",
-		},
+		Wallet:    blockchain.NewWallet(),
 		BlockSize: blockchain.DefaultBlockSize,
 	}
 }
 
-// getCurrentUser gets the current user.
+// getCurrentUser gets the current user function only used for app initalization.
 func getCurrentUser() *User {
 	// TODO: Check for local user information on disk,
 	// If doesnt exist, create new user.
 	return NewUser()
 }
+
+// Pay pays an amount of coin to an address `to`.
+func (a *App) Pay(to string, amount uint64) error {
```

Comment:
I think we need to make some changes to this function.

We should:
1. Create a signed transaction
2. Broadcast it to all connected nodes

Independently of this function, the balance of the current user's wallet should be computed each time a new block is added to our chain.